pylinalg/obj/transform.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+
+from ..func import quaternion as qt
+
+
+class Transform:
+    """
+    Abstract base class to define a common interface and allow future extension
+    to non-linear transformations.
+    """
+
+
+class AffineTransform(Transform):
+    """
+    A affine transformation.
+
+    This transform represents an affine mapping between two coordinate frames:
+    parent and child. That is, a transformation that is linear in homogeneous
+    coordinates and that consists of a translation, rotation, and scale (applied
+    in that order).
```

Comment:
The order is always SRT (scale, rotate, translate)

Reply:
Yes, I think we are using different semantics here. How about we avoid using `parent` and `child` while documenting this class? `parent` is aliased with the parent object in the scene graph, so it can become confusing when talking about the object's parent (object) vs the transform's parent (frame).